protocol/txsender/sender.go

```diff
@@ -42,6 +34,7 @@ type privateKeySender struct {
 	ec       *ethclient.Client
 	contract *swapfactory.SwapFactory
 	txOpts   *bind.TransactOpts
+	txLock   sync.Mutex // locks from TX start until receipt so we don't reuse ETH nonce values
```

Comment:
This is a new and important change.  If we are doing multiple current swaps, we can't have multiple pending transactions from the same wallet running at the same time.  If there is a use case for doing multiple swaps simultaneously with short timeouts, the user is going to want to be able to use multiple wallets so one swap doesn't block an important timeout for a different swap while a transaction is getting processed.